planner/core/logical_plan_builder.go

```diff
@@ -1353,6 +1353,9 @@ func (b *PlanBuilder) buildProjection4Union(ctx context.Context, u *LogicalUnion
 			childTp := u.children[j].Schema().Columns[i].RetType
 			resultTp = unionJoinFieldType(resultTp, childTp)
 		}
+		if err := expression.CheckIllegalMixCollation("UNION", tmpExprs, types.ETInt); err != nil {
```

Comment:
Why use types.ETInt here?

Reply:
since `union` will compare the data to remove the duplicate one, and the comparison will return `1` which is `int`